src/main/java/uk/gov/hmcts/ccd/domain/model/definition/CaseField.java

```diff
@@ -140,6 +144,46 @@ public void propagateACLsToNestedFields() {
         propagateACLsToNestedFields(this, this.accessControlLists);
     }
 
+    @JsonIgnore
+    public FieldType getFieldTypeByPath(String path) {
```

Comment:
I might be wrong but such a method would be better placed in the FieldType class rather than CaseField. You could return an Optional<FieldType>. And the following check:
`if (StringUtils.isBlank(path)) {
            return this.getFieldType();
 }`
could be removed from here and placed in the `DefaultFindSearchInputOperation`

Reply:
Moved to CaseType. Now it is named getCaseFieldByPath(String caseFieldId, String path)